agent/consul/server.go

```diff
@@ -1010,20 +1010,50 @@ func (s *Server) WANMembers() []serf.Member {
 }
 
 // RemoveFailedNode is used to remove a failed node from the cluster
-func (s *Server) RemoveFailedNode(node string) error {
+func (s *Server) RemoveFailedNode(node string, prune bool) error {
+	if prune {
+		s.removeFailedNodePrune(node)
```

Comment:
Did you mean to `return s.removeFailedNodePrune(node)` ?

Reply:
If so, then this makes more sense, and then I have a suggestion if you wanted to cut down on boilerplate more:

```
func (s *Server) RemoveFailedNode(node string, prune bool) error {
	var removeFn func(*serf.Serf, string) error
	if prune {
		removeFn = (*serf.Serf).RemoveFailedNodePrune
	} else {
		removeFn = (*serf.Serf).RemoveFailedNode
	}

	if err := removeFn(s.serfLAN, node); err != nil {
		return err
	}
	// The Serf WAN pool stores members as node.datacenter
	// so the dc is appended if not present
	if !strings.HasSuffix(node, "."+s.config.Datacenter) {
		node = node + "." + s.config.Datacenter
	}
	if s.serfWAN != nil {
		if err := removeFn(s.serfWAN, node); err != nil {
			return err
		}
	}
	return nil
}
```

Reply:
I <3 this, thank you